### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by manuelbuil**
Based on the github actions in the sriov-operator project, add the same kind of actions here.

We did not have a non-rhel Dockerfile in this project, so I added one

Signed-off-by: Manuel Buil <mbuil@suse.com>

### CODE REVIEW

**Comment by manuelbuil** [comment]
There was a non-rhel Dockerfile in images, so we will use that one instead

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.15
```
Comment (martinkennelly) [inline .github/workflows/go.yml:15]: What does the '^' symbol accomplish?
Latest golang is 1.16.
Also, do you think there's value iterating over a number of golang versions? 
I have caught a build error before when I have built on newer golang versions. I was hoping to keep a list of the last 4 minor golang versions e.g. 1.16.x, 1.15.x, etc to ensure any changes still build at least across different golang versions. The builds are in parallel so it doesn't effect the speed of the workflow.

```
jobs:
  build:
    strategy:
      matrix:
        go-version: [1.13.x, 1.14.x, 1.15.x, 1.16.x]
        os: [ubuntu-latest]
````

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: fmt
```
Comment (martinkennelly) [inline .github/workflows/go.yml:20]: Is there any need for this task when ```make all``` performs 'fmt' anyway?

File: .github/workflows/image-push-release.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Docker meta
+        id: docker_meta
+        uses: crazy-max/ghaction-docker-meta@v1
+        with:
+          images: ${{ env.IMAGE_NAME }}
+          tag-latest: false
+
+      - name: Build and push sriov-network-operator
```
Comment (martinkennelly) [inline .github/workflows/image-push-release.yml:32]: ```sriov-network-operator``` -> ```sriov-network-device-plugin```

File: .github/workflows/go.yml (hunk 1 above)
Comment (manuelbuil) [reply .github/workflows/go.yml:15]: In theory `^` is regex for "Start with", i.e. ^1.15 should match 1.15.x.

Your idea is good! I saw it in https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pull/332/files, I am just wondering if it works with the "x", can you please confirm it?

File: .github/workflows/go.yml (hunk 2 above)
Comment (manuelbuil) [reply .github/workflows/go.yml:20]: good point, thanks. It makes sense to remove it

File: .github/workflows/image-push-release.yml (hunk 3 above)
Comment (manuelbuil) [reply .github/workflows/image-push-release.yml:32]: thanks!

File: .github/workflows/go.yml (hunk 1 above)
Comment (martinkennelly) [reply .github/workflows/go.yml:15]: Sure - See execution of multiple golang versions here: https://github.com/martinkennelly/network-resources-injector/actions/runs/627385214

**Comment by adrianchiris** [comment]
Hmm, i see #328 #329 #332  address build and test, so perhaps we should limit this to just the image push parts ?
@martinkennelly @manuelbuil  WDYT?

About the PR labeler workflow, i think it may need additional inputs from @zshi-redhat @pliurh if they found it useful in network operator.

**Comment by manuelbuil** [comment]
> Hmm, i see #328 #329 #332 address build and test, so perhaps we should limit this to just the image push parts ?
> @martinkennelly @manuelbuil WDYT?
> 
> About the PR labeler workflow, i think it may need additional inputs from @zshi-redhat @pliurh if they found it useful in network operator.

Ok! Let me just do the image push parts

**Review by martinkennelly — APPROVED** [review]
Thank you for the changes!

**Comment by pliurh** [comment]
LGTM

**Pull request merged**